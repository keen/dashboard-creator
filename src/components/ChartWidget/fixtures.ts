export const chartData = [
  {
    name: 'January',
    users: 3,
    licenses: 52,
    shops: 12,
    books: 34,
  },
  {
    name: 'February',
    users: 6,
    licenses: 54,
    shops: 34,
    books: 89,
  },
  {
    name: 'March',
    users: 20,
    licenses: 15,
    shops: 23,
    books: 41,
  },
  {
    name: 'April',
    users: 19,
    licenses: 82,
    shops: 15,
    books: 23,
  },
];
