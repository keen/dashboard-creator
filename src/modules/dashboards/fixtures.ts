import { DashboardMetaData } from './types';

export const dashbordsMeta: DashboardMetaData[] = [
  {
    id: '@dashboard/01',
    widgets: 5,
    queries: 0,
    title: 'Dashboard 1',
    isPublic: true,
  },
  {
    id: '@dashboard/02',
    widgets: 0,
    queries: 2,
    title: 'Dashboard 2',
    isPublic: true,
  },
];
