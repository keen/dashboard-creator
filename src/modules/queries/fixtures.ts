/* eslint-disable @typescript-eslint/naming-convention */

import { SavedQuery, SavedQueryAPIResponse } from './types';

export const savedQueriesResponse: SavedQueryAPIResponse[] = [
  {
    query: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'logins',
      order_by: null,
    },
    refresh_rate: 0,
    query_name: '@query/01',
    metadata: {
      display_name: 'Query 01',
      visualization: {
        type: 'bar',
        chart_settings: {},
        widget_settings: {},
      },
    },
  },
  {
    query: {
      analysis_type: 'sum',
      timeframe: 'this_14_days',
      event_collection: 'purchases',
      order_by: null,
    },
    refresh_rate: 0,
    query_name: '@query/02',
    metadata: {
      display_name: 'Query 02',
      visualization: {
        type: 'area',
        chart_settings: {},
        widget_settings: {},
      },
    },
  },
];

export const savedQueries: SavedQuery[] = [
  {
    id: '@query/01',
    displayName: 'Query 01',
    visualization: {
      type: 'bar',
      chartSettings: {},
      widgetSettings: {},
    },
    settings: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'purchases',
      order_by: null,
    },
  },
  {
    id: '@query/02',
    displayName: 'Query 02',
    visualization: {
      type: 'metric',
      chartSettings: {},
      widgetSettings: {},
    },
    settings: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'pageviews',
      order_by: null,
    },
  },
];
