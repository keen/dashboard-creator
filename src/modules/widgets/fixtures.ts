import { WidgetItem } from './types';

export const widget: WidgetItem = {
  data: null,
  error: null,
  isConfigured: false,
  isInitialized: false,
  isActive: false,
  isLoading: false,
  isHighlighted: false,
  isFadeOut: false,
  isTitleCover: false,
  widget: {
    id: '@widget/id',
    position: { x: 3, y: 2, w: 6, h: 2 },
    query: null,
    datePickerId: null,
    settings: {
      chartSettings: {},
      visualizationType: null,
      widgetSettings: {},
    },
    type: 'visualization',
  },
};
