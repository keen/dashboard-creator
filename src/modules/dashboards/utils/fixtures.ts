import { DashboardMetaData } from '../../../modules/dashboards';

export const dashboardsMeta: DashboardMetaData[] = [
  {
    id: '@dashboard/01',
    widgets: 5,
    queries: 0,
    title: 'Dashboard 1',
    tags: ['tag-1', 'tag-2', 'tag-4'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/1',
  },
  {
    id: '@dashboard/02',
    widgets: 0,
    queries: 2,
    title: 'Dashboard 2',
    tags: ['tag-2', 'tag-3'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/2',
  },
  {
    id: '@dashboard/03',
    widgets: 0,
    queries: 0,
    title: null,
    tags: ['tag-1', 'tag-4'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/3',
  },
];
