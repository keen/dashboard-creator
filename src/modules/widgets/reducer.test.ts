import widgetsReducer, { initialState } from './reducer';

import {
  createWidget,
  removeWidget,
  setWidgetState,
  setWidgetLoading,
  updateWidgetsPosition,
  finishChartWidgetConfiguration,
} from './actions';
import { widget } from './fixtures';

const gridPosition = { x: 1, y: 2, w: 4, h: 2 };

test('creates visualization widget', () => {
  const action = createWidget('@widget/id', 'visualization', gridPosition);
  const { items } = widgetsReducer(initialState, action);

  expect(items).toMatchSnapshot();
});

test('removes widget', () => {
  const state = {
    ...initialState,
    items: {
      '@widget/id': widget,
    },
  };

  const action = removeWidget('@widget/id');
  const { items } = widgetsReducer(state, action);

  expect(items['@widget/id']).toBeUndefined();
});

test('finishes visualization widget configuration', () => {
  const state = {
    ...initialState,
    items: {
      '@widget/id': widget,
    },
  };

  const query = 'purchases';
  const chartSettings = { layout: 'vertical' };

  const action = finishChartWidgetConfiguration(
    '@widget/id',
    query,
    'bar',
    chartSettings,
    {}
  );
  const {
    items: { ['@widget/id']: widgetItem },
  } = widgetsReducer(state, action);

  expect(widgetItem).toMatchSnapshot();
});

test('set widget loading state', () => {
  const state = {
    ...initialState,
    items: {
      '@widget/id': widget,
    },
  };

  const action = setWidgetLoading('@widget/id', true);
  const {
    items: { ['@widget/id']: widgetItem },
  } = widgetsReducer(state, action);
  const { isLoading } = widgetItem;

  expect(isLoading).toEqual(true);
});

test('updates widget position', () => {
  const state = {
    ...initialState,
    items: {
      '@widget/id': widget,
    },
  };

  const action = updateWidgetsPosition([{ ...gridPosition, i: '@widget/id' }]);
  const {
    items: { ['@widget/id']: widgetItem },
  } = widgetsReducer(state, action);
  const {
    widget: { position },
  } = widgetItem;

  expect(position).toEqual(gridPosition);
});

test('set widget state', () => {
  const state = {
    ...initialState,
    items: {
      '@widget/id': widget,
    },
  };

  const action = setWidgetState('@widget/id', {
    isConfigured: true,
    isInitialized: true,
  });
  const {
    items: { ['@widget/id']: widgetItem },
  } = widgetsReducer(state, action);

  const { isConfigured, isInitialized } = widgetItem;

  expect(isConfigured).toEqual(true);
  expect(isInitialized).toEqual(true);
});
