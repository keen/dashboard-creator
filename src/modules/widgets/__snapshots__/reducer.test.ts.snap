// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates visualization widget 1`] = `
Object {
  "@widget/id": Object {
    "data": null,
    "error": null,
    "isConfigured": false,
    "isInitialized": false,
    "isLoading": false,
    "widget": Object {
      "id": "@widget/id",
      "position": Object {
        "h": 2,
        "w": 4,
        "x": 1,
        "y": 2,
      },
      "query": null,
      "settings": Object {
        "chartSettings": Object {},
        "visualizationType": null,
        "widgetSettings": Object {},
      },
      "type": "visualization",
    },
  },
}
`;

exports[`finishes visualization widget configuration 1`] = `
Object {
  "data": null,
  "error": null,
  "isConfigured": true,
  "isInitialized": false,
  "isLoading": false,
  "widget": Object {
    "id": "@widget/id",
    "position": Object {
      "h": 2,
      "w": 6,
      "x": 3,
      "y": 2,
    },
    "query": "purchases",
    "settings": Object {
      "chartSettings": Object {
        "layout": "vertical",
      },
      "visualizationType": "bar",
      "widgetSettings": Object {},
    },
    "type": "visualization",
  },
}
`;
